### PR TITLE
Implement GDS Saving with FileIO Interface

### PR DESCRIPTION
**Pull request by SBuercklin**
Adds `Devices.GDS` as a saver for `format"GDS"` files to `FileIO.jl`

### CODE REVIEW

**Comment by SBuercklin** [comment]
This makes all tests pass on our related internal package which depend on exporting GDS files

File: src/backends/gds.jl [hunk 1]
```diff
@@ -10,7 +10,7 @@ import ..Rectangles: Rectangle
 import ..Polygons: Polygon
 using ..Cells
 
-import FileIO: File, @format_str, load, save, stream, magic, skipmagic
+import FileIO: File, @format_str, load, stream, magic, skipmagic
```
Comment (platawiec) [inline src/backends/gds.jl]: Can we also do the same for `load`? My understanding is that the FileIO interface also requires that we don't implement `FileIO.load`

Comment (SBuercklin) [reply src/backends/gds.jl]: We should be able to. Do we ever end up loading GDS files?

Comment (platawiec) [reply src/backends/gds.jl]: Yes, it's useful to have.

File: Project.toml [hunk 2]
```diff
@@ -19,3 +19,4 @@ Unitful = "1986cc42-f94f-5a68-af5c-568840ba703d"
 
 [compat]
 Clipper = "0.6.0"
+FileIO = "1.11"
```
Comment (platawiec) [inline Project.toml]: I think there should be a compat bound for FileIO, but I'm wondering if this is too aggressive? Is it possible that `FileIO = "1"` would work with these changes? I'm not suggesting an exhaustive search, but maybe check v1, v1.4, and v1.6 (the FileIO docs suggest that 1.6 should work)

Comment (SBuercklin) [reply Project.toml]: 1.6 works, 1.5 does not. `1.6 - 1` sets the half-closed interval `[1.6, 2.0)` which I think is what we want

**Pull request merged**